### PR TITLE
1.Reduce space between question title & options & Make spacing a configurable attribute

### LINKED ISSUE #1967: [SDC] Reduce space between question title & options in default layout
**Problem**
The spacing between the question title and options is too big. It is also not configurable for developers so they can easily adjust.

**Solution**
1. Adjust the default spacing to be 4dp (so it is consistent with a 4dp grid)
2. Make spacing a configurable attribute

**EmCare commit**
https://github.com/google/android-fhir/commit/1b9023b3f585418bc5624d8050e886ee878e6706

cc @santosh-pingle @kunjan8794 @khyativyasargus @jingtang10

### PR DESCRIPTION
**Pull request by khyativyasargus**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1967 

**Description**
1. Reduce space between question title & options so that the default spacing to be 4dp (so it is consistent with a 4dp grid)
2. Make spacing a configurable attribute

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Feature

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Comment by khyativyasargus** [comment]
@shelaghm  @santosh-pingle 

**Before SS (16dp padding between each question title and options)**

image



**After default padding to 4dp SS**

image

**Comment by khyativyasargus** [comment]
Hi @shelaghm Thanks for inputs!

**> To my eyes, the 4dp still looks like it is more than 4dp. The 'Gender' padding seems smaller than the other ones? Is there padding on other elements which is making it appear to be more than 4dp?**

I think the screenshots I attached here is in bigger size visually that's why you might be feeling like it's more than 4dp. Let me share proper ss again and let me know on that. 
and for Gender widget I'll check today what's causing it to be different than others.

File: datacapture/src/main/res/values/styles.xml [hunk 1]
```diff
@@ -186,6 +188,10 @@
         <item name="android:textColor">?attr/colorOnSurfaceVariant</item>
     </style>
 
+    <style name="Questionnaire.QuestionHeaderStyle" parent="">
+        <item name="android:paddingBottom">@dimen/header_padding_bottom</item>
+    </style>
```
Comment (santosh-pingle) [inline datacapture/src/main/res/values/styles.xml:193]: Can you please move this at line no. 176 ?

File: datacapture/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -43,7 +43,9 @@
         <item
             name="questionnaireQuestionTextStyle"
         >@style/TextAppearance.Material3.TitleMedium</item>
-
+        <item
+            name="questionnaireQuestionHeaderStyle"
+        >@style/Questionnaire.QuestionHeaderStyle</item>
```
Comment (santosh-pingle) [inline datacapture/src/main/res/values/styles.xml:48]: Can you please move this at line no. 43 just before questionnaireQuestionTextStyle item

File: datacapture/src/main/res/values/attrs.xml [hunk 3]
```diff
@@ -23,6 +23,9 @@
     <!--   Style for question text. -->
     <attr name="questionnaireQuestionTextStyle" format="reference" />
 
+    <!--   Style for question header layout. -->
+    <attr name="questionnaireQuestionHeaderStyle" format="reference" />
+
```
Comment (santosh-pingle) [inline datacapture/src/main/res/values/attrs.xml:28]: nit : Can you please move this at line no. 22 ?

**Comment by santosh-pingle** [comment]
@khyativyasargus 
Can you please also add screenshots for changes when instructions text are also present.

If you see default layout in the catalog app, it has instructions just below the question text.

**Comment by santosh-pingle** [comment]
@khyativyasargus any update on this PR

**Comment by khyativyasargus** [comment]
> instructions

image
image
image

**Comment by khyativyasargus** [comment]
@santosh-pingle As Shelagh observed, Questions with EditTextView as answer view having more space than others due to EditText default style outer space, you can notice in screenshots I shared.

**Review by jingtang10 — APPROVED** [review]

File: datacapture/src/main/res/values/attrs.xml [hunk 4]
```diff
@@ -17,9 +17,15 @@
 <resources>
     <!-- Styles for widgets.   -->
 
+    <!--   Style for group type question header layout. -->
+    <attr name="questionnaireQuestionGroupTypeHeaderStyle" format="reference" />
```
Comment (jingtang10) [inline datacapture/src/main/res/values/attrs.xml:21]:
```suggestion
    <attr name="questionnaireGroupTypeHeaderStyle" format="reference" />
```

File: datacapture/src/main/res/values/styles.xml [hunk 5]
```diff
@@ -37,14 +37,19 @@
         name="Theme.Questionnaire"
         parent="Theme.Material3.DayNight.NoActionBar"
     >
+        <item name="questionnaireQuestionGroupTypeHeaderStyle">
```
Comment (jingtang10) [inline datacapture/src/main/res/values/styles.xml:40]:
```suggestion
        <item name="questionnaireGroupTypeHeaderStyle">
```

File: datacapture/src/main/res/values/styles.xml [hunk 6]
```diff
@@ -172,6 +177,14 @@
         </item>
     </style>
 
+    <style name="Questionnaire.QuestionGroupTypeHeaderStyle" parent="">
```
Comment (jingtang10) [inline datacapture/src/main/res/values/styles.xml:180]:
```suggestion
    <style name="Questionnaire.GroupTypeHeaderStyle" parent="">
```

**Review by jingtang10 — APPROVED** [review]

**Pull request merged**